custom/onse/tasks.py

```diff
@@ -83,9 +85,61 @@ def update_facility_cases_from_dhis2_data_elements(
         otherwise they are emailed.
 
     """
+    _update_facility_cases_from_dhis2_data_elements(period, print_notifications)
+
+
+@task
+def _update_facility_cases_from_dhis2_data_elements(period, print_notifications, retry_attempt=0):
     if not domain_exists(DOMAIN):
         return
     dhis2_server = get_dhis2_server(print_notifications)
+
+    if _server_ready(dhis2_server):
+        _execute_update_facility_cases_from_dhis2_data_elements(dhis2_server, period, print_notifications)
+    else:
+        retry_attempt += 1
+        if retry_attempt <= MAX_RETRY_ATTEMPTS:
+            retry_in_n_days = 2 ** (retry_attempt - 1)
+            retry_date = datetime.utcnow() + relativedelta(days=retry_in_n_days)
+
+            print(f'Importing ONSE ISS facility cases from DHIS2 failed: server unreachable. '
+                  f'Retrying on {retry_date.date()}')
```

Comment:
print. If you want to log this you should use a logger. Also it shouldn't include the hard coded project name. (Also applies to the print below

Reply:
You asking made me realise I wanted to refactor the notification part a bit but I _completely_ forgot. I basically wanted to keep to the file's notification methods (as seen [here](https://github.com/dimagi/commcare-hq/blob/4a64a8d722724d80ddebbceff65e9d7b1290468d/custom/onse/tasks.py#L324)). I have updated this branch with my new code handling the notifications. 

> ...it shouldn't include the hard coded project name...

Are you suggesting deriving it somehow or simply assigning the project name to a variable for a more DRY approach?

Reply:
I came across [this example](https://github.com/dimagi/commcare-hq/blob/71a1618abb58ed341467ffb366313f41fa27d0a4/corehq/motech/requests.py#L187) in the code. It seems the project is derived from the domain. I guess this could work.

Reply:
I'm suggesting deriving it (or just using the domain name)